cmds/core/lock-msrs/lock-msrs.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// lock-msrs locks a bunch of intel MSRs, all specified in the Intel
+// Software developer's manual. This seems like a good set of bits to
+// lock down when booting through NERF/LINUXBOOT to some other OS.
+package main
+
+import (
+	"log"
+
+	"github.com/u-root/u-root/pkg/msr"
+)
+
+const usage = `msr r glob register
+msr w glob register value`
+
+const MSR_IA32_FEATURE_CONTROL   uint32 = 0x3A
+const MSR_PKG_CST_CONFIG_CONTROL uint32 = 0xE2
+const MSR_FEATURE_CONFIG         uint32 = 0x13C
+const MSR_DRAM_POWER_LIMIT       uint32 = 0x618
+const MSR_CONFIG_TDP_CONTROL     uint32 = 0x64B
+const IA32_DEBUG_INTERFACE       uint32 = 0xC80
+
+var msrList = []struct {
```

Comment:
These are Skylake or newer? Or what platforms? 

Can we do cpuid detection and lock what's appropriate? E.g. what about AMD Naples/Rome/etc?

Reply:
Some are architectural, some are Skylake, some are others that are older but not architectural. I figured if the MSR doesn't exist, the kernel will handle the GPF gracefully and we'll get an error I will test it.

Reply:
Invalid MSRs should just be gracefully handled by the kernel. I've added a check for intel only.